lms/static/scripts/frontend_apps/utils/api.ts

```diff
@@ -363,22 +363,21 @@ export function usePaginatedAPIFetch<
   }, [fetchResult.data?.pagination.next, fetchResult.isLoading]);
 
   useEffect(() => {
-    // Every time data is loaded, append the result to the list
-    setCurrentList(prev => {
-      if (!fetchResult.data) {
-        return prev;
-      }
+    if (!fetchResult.data) {
+      return;
+    }
 
-      return [...(prev ?? []), ...fetchResult.data[prop]] as ListType;
-    });
+    const data = fetchResult.data;
+    const hasMorePages = !!data.pagination.next;
 
-    const hasMorePages = !!fetchResult.data?.pagination.next;
+    // Every time data is loaded, append the result to the list
+    setCurrentList(prev => [...(prev ?? []), ...data[prop]] as ListType);
     setHasMorePages(hasMorePages);
```

Comment:
The type of `hasMorePages` is currently a boolean that starts out as `false`, so it doesn't directly represent the initial state where we are fetching the first page and don't know if there will be more afterwards. You can infer this indirectly from `pageSize` being undefined though.

Reply:
Yeah, that's a good point. I spend some time thinking if the initial value should be `true` or `false`, but perhaps you are right and we should in fact initialize it to something else, perhaps `undefined`, meaning "we still don't know"